### PR TITLE
[Search Jobs]: Connect view logs and download results search job UI

### PR DESCRIPTION
**Pull request by vovakulikov**
This PR just connects download result and view logs buttons to the search job API 

## Test plan
- Check that you can download results for search jobs that have download links (all successful search jobs)
- Check that you can download the logs file

### CODE REVIEW

File: client/web/src/enterprise/search-jobs/SearchJobsPage.tsx [hunk 1]
```diff
@@ -318,9 +328,17 @@ const SearchJob: FC<SearchJobProps> = props => {
...
             <span className={styles.jobActions}>
-                <Button variant="link" className={styles.jobViewLogs}>
-                    View logs
-                </Button>
+                <Tooltip content={!job.logURL ? 'There are no logs yet' : ''}>
+                    <DownloadFileButton
+                        variant="link"
+                        disabled={!job.logURL}
+                        fileUrl={job.logURL ?? ''}
+                        fileName={`search-logs-${startDate}-${escapeRegExp(job.query)}`}
```
Comment (stefanhengl) [inline client/web/src/enterprise/search-jobs/SearchJobsPage.tsx:336]: The url is now .api/search/export/3.log (and 3.csv for the results). Should we just determine the filename from the base of the URL?

File: cmd/frontend/internal/httpapi/router/router.go [hunk 2]
```diff
@@ -79,8 +79,8 @@ func New(base *mux.Router) *mux.Router {
 	base.Path("/scip/upload").Methods("POST").Name(SCIPUpload)
 	base.Path("/scip/upload").Methods("HEAD").Name(SCIPUploadExists)
 	base.Path("/search/stream").Methods("GET").Name(SearchStream)
-	base.Path("/search/export/{id}").Methods("GET").Name(SearchJob)
-	base.Path("/search/export/{id}/logs").Methods("GET").Name(SearchJobLogs)
+	base.Path("/search/export/{id}.csv").Methods("GET").Name(SearchJobResults)
+	base.Path("/search/export/{id}.log").Methods("GET").Name(SearchJobLogs)
```
Comment (stefanhengl) [inline cmd/frontend/internal/httpapi/router/router.go:83]: cc @keegancsmith

File: client/web/src/enterprise/search-jobs/SearchJobsPage.tsx (hunk 1 above)
Comment (keegancsmith) [reply client/web/src/enterprise/search-jobs/SearchJobsPage.tsx:336]: agreed we should do it this way. We can update our routes to give this sort of name if its useful. The date in the name is likely useful.

File: cmd/frontend/internal/httpapi/router/router.go (hunk 2 above)
Comment (keegancsmith) [reply cmd/frontend/internal/httpapi/router/router.go:83]: LGTM

**Pull request merged**